plugins/MauticCrmBundle/Integration/SalesforceIntegration.php

```diff
@@ -2032,11 +2032,15 @@ protected function processCompositeResponse($response, &$totalUpdated = 0, &$tot
                     $this->logIntegrationError($exception);
                     $integrationEntity = null;
                     if ($integrationEntityId && $object !== 'CampaignMember') {
-                        $integrationEntity = $this->em->getReference('MauticPluginBundle:IntegrationEntity', $integrationEntityId);
-                        $integrationEntity->setLastSyncDate(new \DateTime());
+                        $integrationEntity = $this->em->find('MauticPluginBundle:IntegrationEntity', $integrationEntityId);
+                        if ($integrationEntity) {
+                            $integrationEntity->setLastSyncDate(new \DateTime());
+                        }
                     } elseif (isset($campaignId) && $campaignId != null) {
-                        $integrationEntity = $this->em->getReference('MauticPluginBundle:IntegrationEntity', $campaignId);
-                        $integrationEntity->setLastSyncDate($this->getLastSyncDate());
+                        $integrationEntity = $this->em->find('MauticPluginBundle:IntegrationEntity', $campaignId);
+                        if ($integrationEntity) {
```

Comment:
This is duplicated code with different parameter - It should be extracted to another class (as commented above) and `setLastSyncDate ` should be encapsulated to that method.